### PR TITLE
Implement QoS 1

### LINKED ISSUE #15: Implement publishing with QoS 1
Implement publishing with QoS 1

### PR DESCRIPTION
**Pull request by JavaSaBr**
Implement QoS 1

### CODE REVIEW

File: src/main/java/com/ss/mqtt/broker/model/MqttPropertyConstants.java [hunk 1]
```diff
@@ -47,4 +50,7 @@
     boolean WILDCARD_SUBSCRIPTION_AVAILABLE_DEFAULT = false;
     boolean SHARED_SUBSCRIPTION_AVAILABLE_DEFAULT = false;
     boolean SUBSCRIPTION_IDENTIFIER_AVAILABLE_DEFAULT = false;
+
+    int PACKET_ID_FOR_QOS_0 = 0;
+    int PACKET_ID_MAX = 0xFFFF;
```
Comment (crazyrokr) [inline src/main/java/com/ss/mqtt/broker/model/MqttPropertyConstants.java]: Is MAXIMUM_PACKET_ID the same ?

File: src/main/java/com/ss/mqtt/broker/model/impl/DefaultMqttSession.java [hunk 2]
```diff
@@ -1,22 +1,170 @@
...
+    }
+
+    @Override
+    public void removeExpiredPackets() {
+
+        if (pendingPublishes.isEmpty()) {
+            return;
+        }
+
+        pendingPublishes.runInWriteLock(publishes -> {
```
Comment (crazyrokr) [inline src/main/java/com/ss/mqtt/broker/model/impl/DefaultMqttSession.java]: please extract to method

File: src/main/java/com/ss/mqtt/broker/model/impl/DefaultMqttSession.java [hunk 3]
```diff
@@ -1,22 +1,170 @@
...
+                    length--;
+                }
+            }
+        });
+    }
+
+    @Override
+    public void resendPendingPacketsAsync(@NotNull MqttClient client, int retryInterval) {
+        var currentTime = System.currentTimeMillis();
+        var stamp = pendingPublishes.readLock();
```
Comment (crazyrokr) [inline src/main/java/com/ss/mqtt/broker/model/impl/DefaultMqttSession.java:120]: runInReadLock can be used

File: src/main/java/com/ss/mqtt/broker/service/impl/InMemoryMqttSessionService.java [hunk 4]
```diff
@@ -65,10 +66,10 @@ public InMemoryMqttSessionService(int cleanInterval) {
     @Override
     public @NotNull Mono<Boolean> store(@NotNull String clientId, @NotNull MqttSession session, long expiryInterval) {
```
Comment (crazyrokr) [inline src/main/java/com/ss/mqtt/broker/service/impl/InMemoryMqttSessionService.java]: Please make multi line

**Review by crazyrokr — CHANGES_REQUESTED** [review]

File: src/main/java/com/ss/mqtt/broker/model/impl/DefaultMqttSession.java (hunk 3 above)
Comment (JavaSaBr) [reply src/main/java/com/ss/mqtt/broker/model/impl/DefaultMqttSession.java:120]: we don't have a version of this method with one object and one primitive argument

File: src/main/java/com/ss/mqtt/broker/model/impl/DefaultMqttSession.java [hunk 5]
```diff
@@ -167,4 +138,32 @@ public void updatePendingPacket(
     public void clear() {
         pendingPublishes.runInWriteLock(Collection::clear);
     }
+
+    private void removeExpiredPackets(@NotNull Array<PendingPublish> publishes) {
```
Comment (crazyrokr) [inline src/main/java/com/ss/mqtt/broker/model/impl/DefaultMqttSession.java]: It can be static

**Review by crazyrokr — CHANGES_REQUESTED** [review]

**Review by crazyrokr — APPROVED** [review]

**Pull request merged**